pygmt/helpers/utils.py

```diff
@@ -223,45 +230,54 @@ def data_kind(
     >>> import numpy as np
     >>> import xarray as xr
     >>> import pathlib
+    >>> [data_kind(data=data) for data in (2, 2.0, True, False)]
+    ['arg', 'arg', 'arg', 'arg']
     >>> data_kind(data=None)
     'vectors'
-    >>> data_kind(data=np.arange(10).reshape((5, 2)))
-    'matrix'
+    >>> data_kind(data=None, required=False)
+    'arg'
     >>> data_kind(data="my-data-file.txt")
     'file'
     >>> data_kind(data=pathlib.Path("my-data-file.txt"))
     'file'
-    >>> data_kind(data=None, required=False)
-    'arg'
-    >>> data_kind(data=2.0, required=False)
-    'arg'
-    >>> data_kind(data=True, required=False)
-    'arg'
+    >>> data_kind(data=["data1.txt", "data2.txt"])
+    'file'
     >>> data_kind(data=xr.DataArray(np.random.rand(4, 3)))
     'grid'
     >>> data_kind(data=xr.DataArray(np.random.rand(3, 4, 5)))
     'image'
+    >>> data_kind(data=np.arange(10).reshape((5, 2)))
+    'matrix'
+    >>> data_kind(data=[1, 2, 3])
+    'matrix'
     """
-    kind: Literal["arg", "file", "geojson", "grid", "image", "matrix", "vectors"]
+    # data is None, so data must be given via a series of vectors (i.e., x/y/z).
+    # The only exception is when dealing with optional virtual files.
+    if data is None:
+        return "vectors" if required else "arg"
+
+    # A file or a list of files
     if isinstance(data, str | pathlib.PurePath) or (
-        isinstance(data, list | tuple)
```

Comment:
Check if `data` is a non-string iterator, instead of just checking if data is a list or a tuple.